File: Interfaces/IQuotationService.cs
using cms_webapi.DTOs;
using cms_webapi.Models;

namespace cms_webapi.Interfaces
{
    public interface IQuotationService
    {
        Task<IEnumerable<QuotationGetDto>> GetAllQuotationsAsync();
        Task<QuotationGetDto?> GetQuotationByIdAsync(long id);
        Task<QuotationDto> CreateQuotationAsync(CreateQuotationDto createQuotationDto);
        Task<QuotationDto?> UpdateQuotationAsync(long id, UpdateQuotationDto updateQuotationDto);
        Task<bool> DeleteQuotationAsync(long id);
        Task<IEnumerable<QuotationGetDto>> GetQuotationsByPotentialCustomerIdAsync(long potentialCustomerId);
        Task<IEnumerable<QuotationGetDto>> GetQuotationsByRepresentativeIdAsync(long representativeId);
        Task<IEnumerable<QuotationGetDto>> GetQuotationsByStatusAsync(int status);
        Task<bool> QuotationExistsAsync(long id);
    }
}

File: WebApi/Interfaces/IQuotationService.ts
import { QuotationDto, CreateQuotationDto, UpdateQuotationDto, QuotationGetDto } from '../Models/QuotationDto';

export interface IQuotationService {
  getAllQuotations(): Promise<QuotationGetDto[]>;
  getQuotationById(id: number): Promise<QuotationGetDto | null>;
  createQuotation(createQuotationDto: CreateQuotationDto): Promise<QuotationDto>;
  updateQuotation(id: number, updateQuotationDto: UpdateQuotationDto): Promise<QuotationDto | null>;
  deleteQuotation(id: number): Promise<boolean>;
  getQuotationsByPotentialCustomerId(potentialCustomerId: number): Promise<QuotationGetDto[]>;
  getQuotationsByRepresentativeId(representativeId: number): Promise<QuotationGetDto[]>;
  getQuotationsByStatus(status: number): Promise<QuotationGetDto[]>;
  quotationExists(id: number): Promise<boolean>;
}